tools/internal_ci/helper_scripts/prepare_build_macos_rc

```diff
@@ -48,6 +48,10 @@ set -ex
 
 # cocoapods
 export LANG=en_US.UTF-8
+# pre-fetch cocoapods master repo with HEAD only
+mkdir -p ~/.cocoapods/repos
+git clone --depth 1 https://github.com/CocoaPods/Specs.git ~/.cocoapods/repos/master
```

Comment:
let's add `time` to see how long the clone takes. Btw, it this supposed to reduce the time needed for the next command "pod repo update" or is this for a subsequent step sometimes later in objc tests?

Reply:
From the comment on the PR is seems this is speeding up ObjC's `build_tests.sh` and not the `pod repo update` command. The speedup is great, but I'm confused here.

From the logs it looks like the first thing "pod repo update" does is *Updating spec repo `master`
Performing a deep fetch of the `master` specs repo to improve future performance* by running the command `/usr/local/bin/git -C /Users/kbuilder/.cocoapods/repos/master fetch origin --progress`, which seems to take 9 minutes.

So it looks like that even when we first clone only what we need, the `pod repo update` command fetches everything anyway right after that.  In which case  it seems that the first command should basically no effect, in which case I don't understand how `build_tests.sh` could actually be getting sped up. Something doesn't add up here IMHO.

Reply:
What's even more odd is that the 9min run of `pod repo update` seems to be triggered by the newly added git clone command - I looked at the tests on master and the usual duration of `pod repo update` is only a few seconds. And such regression is definitely not ok (partly because it would also slow down all other non-objc mac tests by 9mins).

Reply:
As commented above, I suggest removing this line. Line 53 and 55 do not work with each other.

Reply:
We can remove that line, but I'd like to understand how this works. It seems that having a shallow vs deep clone influence what later happens in `pod install` (which seems to be also cloning or updating the repo under some circumstances) and it has big impact on the duration of build_tests.sh - and without understanding this, I don't think we can actually fix this well
(in a way that would be effective and will continue working in the future and will be maintainable).

Reply:
I have a vague memory that after a certain version of Cocoapods, `pod install` no longer pulls the master repo (and it now has an option `--repo-update` to force that). I did not find any reference to that right now though; will post it later. 

Assuming this is true, then this PR helps because:
* In the old script, `pod repo update` acted like no-op because there is no repo exist in directory `~/.cocoapods/repos/master` (because it's a fresh mac image). The actual downloading of master repo is put off into `build_tests.sh` when it invokes `pod install` and took about 12 minutes to finish.
* In the new script, `git clone --depth 1 ...` command fetches the latest commit of the master repo into `~/.cocoapods/repos/master`. Later in `build_tests.sh`, `pod install` command does not force a `pod repo update`, so it just uses the pods in the latest commit of master repo, saved the 12 minutes of download of full repo.